torchvision/datasets/eurosat.py

```diff
@@ -0,0 +1,72 @@
+import os
+from typing import Any
+
+from .folder import ImageFolder
+from .utils import download_and_extract_archive
+
+
+class EuroSAT(ImageFolder):
+    """RGB version of the `EuroSAT <https://arxiv.org/pdf/1709.00029.pdf>`_ Dataset.
+
+    Args:
+        root (string): Root directory of dataset where ``EuroSAT.zip`` exists.
+        download (bool, optional): If True, downloads the dataset from the internet and
+            puts it in root directory. If dataset is already downloaded, it is not
+            downloaded again.
+        transform (callable, optional): A function/transform that  takes in an PIL image
+            and returns a transformed version. E.g, ``transforms.RandomCrop``
+        target_transform (callable, optional): A function/transform that takes in the
+            target and transforms it.
+    """
+
+    url = "https://madm.dfki.de/files/sentinel/EuroSAT.zip"
+    md5 = "c8fa014336c82ac7804f0398fcb19387"
+
+    _class_map = {
+        "AnnualCrop": "Annual Crop",
+        "HerbaceousVegetation": "Herbaceous Vegetation",
+        "Industrial": "Industrial Buildings",
+        "PermanentCrop": "Permanent Crop",
+        "Residential": "Residential Buildings",
+        "SeaLake": "Sea & Lake",
+    }
```

Comment:
Sorry, this has possibly been discussed already, but what's the reason for changing the class names? At first sight I would say we should stick to the initial names?

Reply:
See https://github.com/pytorch/vision/pull/5114#discussion_r771967671.

Reply:
Thanks for the ref.

I would prefer sticking to the author's implementation. There seem to be little value in changing the names, and any deviation from the the original folder names is a potential source of confusion for users.

Reply:
I would argue that the author's will is expressed in more detail in the paper, and though the missing spaces are not an issue for understanding, the truncated class names could be. Rather than a deviation, I'd say it's more an addition to stick closer to the paper, which is more comprehensive :ok_hand: 

But if you prefer to change that, let me know!

Reply:
I agree with @frgfm here. Ideally, a user should be able to use our interface to the data without ever having seen the raw data. Thus, I would prefer to stick closer to the paper rather than the data. 

In general, the classes should be human readable. Of course `SeaLake` is still readable, but `Sea & Lake` conveys the meaning better. In most cases these changes only are made since the classes are also folder names and spaces or symbols shouldn't be part of them.

Reply:
@pmeier are there other instances where we have changed the class names for similar reasons?

In COCO, we have kept names such as `pottedplant`, `tvmonitor`, or `diningtable`.

Personally I don't think the current names are confusing, nor ambiguous: AnnualCrop  Forest  HerbaceousVegetation  Highway  Industrial  Pasture	PermanentCrop  Residential  River  SeaLake

There are cases where users need the path of the files they use. This is not supported in torchvision but this has been asked various times and this is a typical use-case for e.g. submitting to datasets leaderboards. So often, users will hack around the class to return the path and do what they want to do. Any discrepancy between the class names and the path is a potential source of confusion for the user here.

I understand it may sound overcautious (and maybe it is), but every decision that we make introduces risks. I see no risk in keeping the current class names, but I believe there are potential pitfalls with changing them, even if we haven't clearly identified them yet.

Reply:
> are there other instances where we have changed the class names for similar reasons?

I don't recall an instance of the top of my head.

> Personally I don't think the current names are confusing, nor ambiguous

If it is just a space or a symbol that is missing, I agree. Note that there are 2 categories that drop the word "Building":

- "Industrial Building" -> "Industrial"
- "Residental Building" -> "Residental"

I'm not too familiar with the data, but I'd wager a guess this could also be confusing.

> Returning paths is possibly something that we'll support in the future in the new datasets, so I want to avoid setting a precedent of deviating from that here.

The prototype datasets already do that. There are two distinct features usually called "category" and "path" that may or may not carry the same information (of course the will always correspond to the same information). 

---

Anyway, as @frgfm already stated in https://github.com/pytorch/vision/pull/5114#discussion_r771972043, this is probably is nitpicking. If you want to have your way, I'm not going to oppose it.

Reply:
@NicolasHug and I discussed more offline. We should go with the safest option here and solve this with the prototype datasets. @frgfm could you remove the name mapping? After that (and the merge conflicts), the PR should be good to go.

Reply:
Oops my bad, I think I check the notification right before the last reply of Philip!
I just updated the PR accordingly :+1: